src/Core/AbstractApi.php

```diff
@@ -109,15 +109,17 @@ private function fillEndpoint(?string $endpoint): string
     /**
      * Fallback function for getting the endpoint. This could be overridden by any APIClient.
      *
-     * @param array $uri   parameters that should go in the URI
+     * @param string $uri   parameters that should go in the URI
      * @param array $query parameters that should go in the query string
      */
-    protected function getEndpoint(array $uri, array $query):?string
+    protected function getEndpoint(string $uri, array $query):?string
```

Comment:
There is difference between `getEndpoint` and `fillEndpoint`. 

I wanted `getEndpoint` to be extended by special apis like s3. They have a custom URL..
But Im not super happy with my solution.

Reply:
I've the feeling that endpoints don't have to be customized (everything is available in API.json file).
I've tested my PR (which removes getEndpoint from Client) anbd both S3 and SQS works.